x-pack/elastic-agent/pkg/agent/install/install_unix.go

```diff
@@ -7,8 +7,59 @@
 
 package install
 
+import (
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/application/paths"
+)
+
 // postInstall performs post installation for unix-based systems.
 func postInstall() error {
 	// do nothing
 	return nil
 }
+
+func checkPackageInstall() bool {
+	if runtime.GOOS != "linux" {
+		return false
+	}
+
+	// NOTE searching for english words might not be a great idea as far as portability goes.
+	// list all installed packages then search for paths.BinaryName?
+	// dpkg is strange as the remove and purge processes leads to the package bing isted after a remove, but not after a purge
+
+	// check debian based systems (or systems that use dpkg)
+	// If the package has been installed, the status starts with "install"
+	// If the package has been removed (but not pruged) status starts with "deinstall"
+	// If purged or never installed, rc is 1
+	if _, err := os.Stat("/etc/dpkg"); err == nil {
+		out, err := exec.Command("dpkg-query", "-W", "-f", "${Status}", paths.BinaryName).Output()
```

Comment:
It would be better to use the package name explicitly instead of the binary name. Even if they are the same at the moment use a specific constant for that would be better, that way if it ever changes.

Could you also use `-s` which will only show the status?

Reply:
The output of `-s` includes more information than what we need:
```
vagrant@ubuntu-focal:~$ dpkg-query -s elastic-agent
Package: elastic-agent
Status: install ok installed
Priority: extra
Section: default
Installed-Size: 427208
Maintainer: <@ace0c0ea54fb>
Architecture: amd64
Version: 8.2.0
Conffiles:
 /etc/elastic-agent/.elastic-agent.active.commit 75bbbe44ef52356bb5e85a458166b1b6
 /etc/elastic-agent/elastic-agent.reference.yml 23aa69e9e641a43d7daaa0975edf0ec6
 /etc/elastic-agent/elastic-agent.yml f169e731b8e4c8a571dca0107954534d
 /etc/init.d/elastic-agent 79b301b56cc62629b9d276d6ae24ca2f
Description: Agent manages other beats based on configuration provided.
License: Elastic-License
Vendor: Elastic
Homepage: https://www.elastic.co/beats/elastic-agent
```
where the `-W -f '${Status}'` flags will give just that Status line

Reply:
@blakerouse, do you know where the package name is defined?

Reply:
I believe it is using the name of the beat for the package name, so the way you have it works. I am just suggesting we use a different constant, with the same value. Just to future proof it that maybe one day they could be different.

Its more of a nitpick and if you prefer they way you have it, that is okay as well.

Reply:
One question, here, I see we are interacting directly with the `CLI`, is there any C-Libary of even `go package` that we can use to abstract the package manager?

Reply:
I'm not finding a c or go library that can query dpkg/rpm for installed packages

Reply:
After a bit of digging we can find that the package name is defined by our Mage files as the `BeatServiceName`, this defaults to the same as `BeatName` for almost all cases (exception being heartbeat).
https://github.com/elastic/beats/blob/237937085a5a7337ba06f1268cfc55cd4b869e31/dev-tools/mage/settings.go#L70-L71
The value is set to the directory name we run mage out of (i.e., x-pack/`elastic-agent`). So I don't think it's a good idea to introduce a new variable for package name within the current codebase as it won't be picked up by our tooling.

Reply:
sound good